microsite/src/main/tut/overview/testing_effects.md

```diff
@@ -6,10 +6,201 @@ title:  "Testing Effects"
 
 # {{page.title}}
 
-There are many approaches to testing functional effects, including using free monads, using tagless-final, and using environmental effects. Although all of this approaches are compatible with ZIO, the simplest and most natural is environmental effects.
+There are many approaches to testing functional effects, including using free monads, using tagless-final, and using environmental effects. Although all of these approaches are compatible with ZIO, the simplest and most natural is _environmental effects_.
 
 This section introduces environmental effects and shows how to write testable functional code using them.
 
+```tut:invisible
+import scalaz.zio._
+import scalaz.zio.console._
+```
+
+# Environments
+
+The ZIO data type has an `R` type parameter, which is used to describe the type of _environment_ required by the effect. 
+
+ZIO effects can access the environment using `ZIO.environment`, which provides direct access to the environment, as a value of type `R`.
+
+```tut:silent
+for {
+  env <- ZIO.environment[Int]
+  _   <- putStrLn(s"The value of the environment is: $env")
+} yield env
+```
+
+The environment does not have to be a primitive value like an integer. It can be much more complex. 
+
+When the environment is a type with fields, then the `ZIO.access` method can be used to access a given part of the environment in a single method call.
+
+```tut:silent
+case class Config(server: String, port: Int)
+
+val z: ZIO[Config, Nothing, String] = 
+  for {
+    server <- ZIO.access[Config, String](_.server)
```

Comment:
How about we give access the same treatment as bracket, so the result type can be inferred?

Reply:
Love it! I think one type param will be required...